Foundation/NSURLSession/NSURLSessionTask.swift

```diff
@@ -597,10 +618,11 @@ fileprivate extension URLSessionTask {
     /// expects.
     ///
     /// - SeeAlso: https://curl.haxx.se/libcurl/c/CURLOPT_HTTPHEADER.html
-    func curlHeaders(for request: URLRequest) -> [String] {
+    func curlHeaders(for httpHeaders: [AnyHashable : Any]?) -> [String] {
         var result: [String] = []
         var names = Set<String>()
-        if let hh = currentRequest?.allHTTPHeaderFields {
+        if httpHeaders != nil {
+            let hh = httpHeaders as! [String:String]
```

Comment:
Wouldn't it be better to use an actual `[AnyHashable : Any]` dictionary and turn it's keys and values into strings using `String(describing:)`? 
This would also require `var names = Set<AnyHashable>()`, but `AnyHashable` can handle strings.